draft-enghardt-panrg-path-properties.md

```diff
@@ -84,7 +84,7 @@ in Path-Aware Networking {{I-D.irtf-panrg-questions}}, which is a product of the
 # Terminology
 
 Node:
-: A device that implements a protocol on a specific layer, e.g., the physical layer, the link layer, the network layer, the transport layer, or the application layer.
+: A device that implements a protocol on a specific layer.
```

Comment:
I would refrain from defining a node as a device since in the case of network service chaining, each service would be considered a node, which means they would run on the same device.

Maybe we could replace "device" with "entity".

Another point that may be problematic is the restriction on a (single) specific layer, since some protocols might not clearly be part of a single layer (e.g., ICMP which is a layer 3 protocol but looks like a layer 4 protocol...). Should we maybe simply define a node as: "An entity that implements a function that takes packet(s) as inputs and outputs packet(s)" or is this too abstract?

Reply:
Agree that defining a node as a device is a problem. Similar to links, nodes should not be limited to physical entities.
About the layering, yes. Maybe all this layering complexity in our definitions is unnecessary anyway and we should get rid of it? I think there was a reason to introduce the layering in the first place, but I forgot.

In your proposed definition, you replace "protocol" with a "function", I think that's a good idea - not every node may implement a protocol in the traditional way (e.g., does a firewall implement a protocol if it filters based on headers of this protocol? I wouldn't say so.).
However, "input and output" sounds to me like a node necessarily transforms packets, but it may only send or only receive packets.
So how about "An entity that implements a function which sends and/or receives packets"?
Or maybe even just "An entity that sends and/or receives packets"? Because that's what makes the node different from the link, right?

Reply:
We could say:
An entity that can receive packets from an incoming link and can send arbitrary packets on an outgoing link.

Does this imply that nodes process packets only in one direction? Or is it clear?

Reply:
I went with a more open definition and just say "process" now. What do you think?